From david-b@pacbell.net Tue May 30 20:49:37 2006
From: David Brownell <david-b@pacbell.net>
To: Greg KH <greg@kroah.com>
Subject: USB: cdc_ether: recognize olympus r1000 (fix regression)
Date: Tue, 30 May 2006 20:49:29 -0700
Cc: Christian Herrmann <c.herrmann@ccxl.de>, Todd Blumer <todd@sdgsystems.com>
Message-Id: <200605302049.30269.david-b@pacbell.net>

Earlier work splitting the "usbnet" driver out into a core plus driver
modules was missing a blacklist entry for the Olympus R-1000; it must
not use the CDC Ethernet driver, only the "zaurus" support works with
it.

Signed-off-by: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/net/cdc_ether.c |   12 ++++++++++++
 1 file changed, 12 insertions(+)

--- gregkh-2.6.orig/drivers/usb/net/cdc_ether.c
+++ gregkh-2.6/drivers/usb/net/cdc_ether.c
@@ -455,6 +455,18 @@ static const struct usb_device_id	produc
 	.driver_info		= 0,
 },
 
+/* Olympus has some models with a Zaurus-compatible option.
+ * R-1000 uses a FreeScale i.MXL cpu (ARMv4T)
+ */
+{
+	.match_flags    =   USB_DEVICE_ID_MATCH_INT_INFO
+		 | USB_DEVICE_ID_MATCH_DEVICE,
+	.idVendor               = 0x07B4,
+	.idProduct              = 0x0F02,	/* R-1000 */
+	ZAURUS_MASTER_INTERFACE,
+	.driver_info		= 0,
+},
+
 /*
  * WHITELIST!!!
  *
